app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -242,15 +244,19 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
 
         components.core.requestInterceptor.setNavigationController(navHost.navController)
 
+        MainScope().launch(Dispatchers.Main) {
```

Comment:
I think you want:
```kotlin
runBlocking {
    navGraphJob.await()
}
```

I think the code you wrote would launch a co-routine that eventually would run on the main thread that is basically a no-op (it would run, suspend on `await`, resume when `await` is done, and end.

Reply:
Yes, also I don't this is safe. You will need to make sure no navigation can happen before this runs to completion, and with this (unchanged) various other parts of the system would need to "join" the job.